xstate-chatbot/nodejs/src/machine/hospital-details.js

```diff
@@ -0,0 +1,1068 @@
+const { assign, actions } = require('xstate');
+const moment = require('moment');
+const dialog = require('./util/dialog');
+const mediaUtil = require('./util/media');
+const config = require('../env-variables');
+const { messages, grammers } = require('./messages/hospital-details');
+const { personService, bedsService } = require('./service/service-loader');
+
+const hospitalFlow = {
+  id: 'hospitalFlow',
+  initial: 'nodalOfficer',
+  onEntry: assign((context, event) => {
+    context.slots.hospital = {};
+    context.slots.previoushospitaldata = {};
+  }),
+  states: {
+    nodalOfficer: {
+      id: 'nodalOfficer',
+      initial: 'process',
+      states: {
+        prompt: {
+          onEntry: assign((context, event) => {
+          }),
+          on: {
+            USER_MESSAGE: 'process',
+          },
+        },
+        process: {
+          onEntry: assign((context, event) => {
+            context.slots.hospital.mobileNumber=context.user.mobileNumber;
+          }),
+          invoke: {
+            src: (context, event) => bedsService.getHospitalById(context.slots.hospital.mobileNumber),
```

Comment:
Rename the function as actually, you are doing
getHospitalsByMobileNumber.

Reply:
changed